File: WorkRequestTracker/WorkRequestTracker/Data/Entities/WorkRequest.cs
namespace WorkRequestTracker.Data.Entities
{
    public class WorkRequest
    {
        public int Id { get; set; }

        public required string Title { get; set; }

        public required string ClientName { get; set; }

        public required string Description { get; set; }

        public required PriorityEnum Priority { get; set; }
        
        public required StatusEnum Status { get; set; }

        public required DateTime DueDate { get; set; }

        public required DateTime CreatedDate { get; set; }
        
        public DateTime? UpdatedDate { get; set; }

        public string? Notes { get; set; }
    }
}

File: UI/src/models/workRequest.ts
export enum PriorityEnum {
  'Unknown' = 0,
  'Low' = 1,
  'Medium' = 2,
  'High' = 3
}

export enum StatusEnum {
  'Unknown' = 0,
  'New' = 10,
  'In Progress' = 20,
  'Blocked' = 30,
  'Completed' = 40
}

export interface WorkRequest {
  id: number;
  title: string;
  clientName: string;
  description: string;
  priority: PriorityEnum;
  status: StatusEnum;
  dueDate: string;
  notes?: string | null;
}

